db/db_impl/db_impl_write.cc

```diff
@@ -130,7 +130,12 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
   assert(!seq_per_batch_ || batch_cnt != 0);
   if (my_batch == nullptr) {
     return Status::Corruption("Batch is nullptr!");
-  } else if (WriteBatchInternal::TimestampsUpdateNeeded(*my_batch)) {
+  } else if (!disable_memtable &&
+             WriteBatchInternal::TimestampsUpdateNeeded(*my_batch)) {
+    // If writing to memtable, then we require the caller to set/update the
+    // timestamps for the keys in the write batch.
+    // Otherwise, it means we are just writing to the WAL, and we allow
+    // timestamps unset for the keys in the write batch.
```

Comment:
Sure, will clarify, and I will explain in this comment as well:
in write-committed prepare phase, we do not need the timestamp in the WAL because these keys are not yet inserted to the memtables yet, and they will be used only for recovery. During recovery, should they be inserted to memtables, we will obtain the timestamp from the commit marker which is also in the WAL.